### PR TITLE
  Override Payments

### PR DESCRIPTION
**Pull request by mabbus**

### CODE REVIEW

File: app/Http/Requests/Deal/AttachOverride.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'start_at' => 'date_format:Y-m-d|required',
+            'end_at' => 'date_format:Y-m-d|required',
+            'amount' => 'numeric|required',
```
Comment (lexod) [inline app/Http/Requests/Deal/AttachOverride.php:20]: specify min/max and total amount of digits after coma

File: app/Http/Tasks/Transactions/Deal/ChargeMerchant.php [hunk 2]
```diff
@@ -64,7 +65,14 @@ private function countTodayTransactions()
     }
 
     private function getPayment() {
-        //TODO : Override payments for a date range
+        $overrides = Overrides::where('deal_id', $this->deal->id)
+            ->whereDate('start_at','<=', Carbon::today()->toDateString())
```
Comment (lexod) [inline app/Http/Tasks/Transactions/Deal/ChargeMerchant.php:69]:
```php
->whereDate('start_at','<=', Carbon::today()->startOfDay())
->whereDate('end_at', '>=', Carbon::today()->endOfDay())
```

**Review by lexod — CHANGES_REQUESTED** [review]

File: app/Http/Requests/Deal/AttachOverride.php (hunk 1 above)
Comment (mabbus) [reply app/Http/Requests/Deal/AttachOverride.php:20]: on amount?

Comment (lexod) [reply app/Http/Requests/Deal/AttachOverride.php:20]: yes

**Pull request merged**